firmware/lib/Kinematics/src/Kinematics.h

```diff
@@ -0,0 +1,45 @@
+#ifndef KINEMATICS_H
+#define KINEMATICS_H
+
+#include "Arduino.h"
+#include "Timer.h"
+#include "Encoder.h"
+#include "LimitSwitch.h"
+
+#define VELOCITY_HOMING 200
+
+enum status{pressed, depressed};
+enum direction{positive, negative};
+enum segment{accelerate, hold, decelerate};
```

Comment:
Should typedef all these enums:
```c
typedef enum { PRESSED, DEPRESSED } Status;
// etc.
```
Also think it would be worthwhile using UPPER_CASE_UNDERSCORES style so they don't look like local variables.